x/globalfee/types/authorization.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	"regexp"
+	"strings"
+
+	sdkErrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// validateMethods validates the method names for the code and contract authorizations
+// valid values are alphanumeric names and '*' which is wildcard for all methods of given codeid/contract
+// e.g "mint", "test_burn", "*", "list2"
+func validateMethods(caMethods []string) error {
+	if len(caMethods) == 0 {
+		return sdkErrors.Wrap(ErrInvalidMethods, "empty methods provided")
+	}
+
+	pattern := `^(\*|[\w]+)$` // only allow method names or "*"
```

Comment:
methods can also be snake case `set_bid`

Reply:
\w matches [a-zA-Z0-9_]. So underscore is taken care of